### PR TITLE
Refactor validation

### PR DESCRIPTION
**Pull request by mattcollier**
Adjust inputs to validators.

### CODE REVIEW

File: lib/rootApi.js [hunk 1]
```diff
@@ -94,31 +95,20 @@ api.add = callbackify(async (actor, options) => {
...
-  if(!(ledgerConfiguration && jsonld.hasValue(
-    ledgerConfiguration, 'type', 'WebLedgerConfiguration'))) {
-    throw new BedrockError(
-      'The provided configuration event is malformed. ' +
-      'An object with `type` set to `WebLedgerConfiguration` does not exist.',
-      'DataError',
-      {httpStatusCode: 400, public: true, ledgerConfiguration});
-  }
+  // perform basic schema validation
+  validator.validateGenesisLedgerConfiguration({ledgerConfiguration});
```
Comment (dlongley) [inline lib/rootApi.js]: Does this need `await`?

File: lib/validator.js [hunk 2]
```diff
@@ -4,53 +4,94 @@
...
   // run all the validators in parallel
-  const validatorReports = [];
+  const promises = [];
   for(const validatorConfig of validatorConfigs) {
-    const report = await _runValidator(input, validatorConfig, {ledgerNode});
-    validatorReports.push(report);
+    promises.push(_runValidator(
+      {ledgerConfig, ledgerNode, validatorConfig, validatorInput}));
   }
+  const validatorReports = await Promise.all(promises);
```
Comment (dlongley) [inline lib/validator.js]: Could do: `await Promise.all(validatorConfigs.map(` instead... but just a matter of style really.

File: lib/rootApi.js (hunk 1 above)
Comment (dlongley) [reply lib/rootApi.js]: It looks like it currently doesn't, but I recommend we make it so.

Comment (mattcollier) [reply lib/rootApi.js]: done.

File: lib/validator.js (hunk 2 above)
Comment (mattcollier) [reply lib/validator.js]: done.

**Comment by mattcollier** [comment]
@dlongley yes, that change is still on deck.  I'm going to merge this to master, but not do a release until we change the validator return value.

**Pull request merged**